src/Api.Rest/Common/Client/RestClientBase.cs

```diff
@@ -124,8 +146,42 @@ public abstract class RestClientBase : IDisposable
 			_Serializer = serializer ?? ObjectSerializer.Default;
 
 			MaxUriLength = maxUriLength;
+			_Timeout = timeout ?? DefaultTimeout;
 
-			BuildHttpClient( timeout );
+			BuildHttpClient();
+		}
+
+		/// <summary>
+		/// Initializes a new instance of the <see cref="RestClientBase"/> class. This constructor exists to support
+		/// rest client creation via <see cref="RestClientBuilder"/>.
+		/// </summary>
+		internal RestClientBase( [NotNull] string endpointName, [NotNull] RestClientSettings settings )
+		{
+			if( endpointName == null )
+				throw new ArgumentNullException( nameof( endpointName ) );
+			if( settings == null )
+				throw new ArgumentNullException( nameof( settings ) );
+
+			ServiceLocation = new UriBuilder( settings.ServerUri )
+			{
+				Path = ( settings.ServerUri.AbsolutePath.Replace( "/DataServiceSoap", "" ) + "/" + endpointName ).Replace( "//", "/" )
+			}.Uri;
+
+			_Timeout = settings.Timeout;
+			_UseProxy = settings.UseSystemProxy;
+			_Chunked = settings.AllowChunkedDataTransfer;
+
+			_CustomHttpMessageHandler = null;
+			_AuthenticationHandler = settings.AuthenticationHandler;
+
+			_DelegatingHandlerFactories = new List<Func<DelegatingHandler>>( settings.DelegatingHandlerFactories );

```

Comment:
Out of curiosity, is there a reason why the factories are given into a new List, instead of directly using the one from the settings?

Reply:
That is a bit of defensive programming. Since the list in the settings can be freely set, anyone with the original instance of the list can modify it. I want the rest client to always use the list items it was constructed with.